source/plugin/text/templates.rst

```diff
@@ -0,0 +1,135 @@
+        arg("server").color(TextColors.RED).style(TextStyles.BOLD), " ",
+        arg("player").color(TextColors.AQUA).style(TextStyles.BOLD), "!"
+    );
+
+TextTemplates, like Text objects themselves are serializable to Configurate. To save a TextTemplate to
+a configuration file use the following code. We are also going to add some settings here so the user can define the
+name of their server.
+
+.. tip::
+
+    To learn more about how to use Configurate to create configuration files for your plugin please refer to
+    :doc:`/plugin/configuration/index`.
+
+.. code-block:: java
+
+    import ninja.leaping.configurate.ConfigurationNode;
+    import com.google.common.reflect.TypeToken;
+
+    ConfigurationNode node = loader.load();
+    node.getNode("serverName").setValue("My Sponge Server");
+    node.getNode("mytemplate").setValue(TypeToken.of(TextTemplate.class), template);
+    loader.save(node);
+
+This will produce the following output:
+
+.. code-block:: guess
+
+    serverName="My Sponge Server"
+    mytemplate {
+        arguments {
+            player {
+                optional=false
+            }
+            server {
+                optional=false
+            }
+        }
+        content {
+            color=yellow
+            extra=[
+                "Welcome to ",
+                {
+                    bold=true
+                    color=red
+                    text="{server}"
+                },
+                " ",
+                {
+                    bold=true
+                    color=aqua
+                    text="{player}"
+                },
+                "!"
+            ]
+            italic=true
+            text=""
+        }
+        options {
+            closeArg="}"
+            openArg="{"
+        }
+    }
+
+You can retrieve TextTemplate's from configurations using the following code:
+
+.. code-block:: java
+
+    TextTemplate template = node.getNode("mytemplate").getValue(TypeToken.of(TextTemplate.class));
+
+Once you are happy with the layout of your new TextTemplate, let's go ahead and send it to the server when a player
+joins the server for the first time. We can achieve that using the following code:
+
+.. code-block:: java
+
+    import com.google.common.collect.ImmutableMap;
+    import org.spongepowered.api.Sponge;
+    import org.spongepowered.api.data.key.Keys;
+    import org.spongepowered.api.entity.living.player.Player;
+    import org.spongepowered.api.event.network.ClientConnectionEvent;
+    import org.spongepowered.api.text.Text;
+    import java.time.Instant;
+    import java.util.Optional;
+
+    @Listener
+    public void onJoin(ClientConnectionEvent.Join event) {
+        Player player = event.getTargetEntity();
+        Instant firstPlayed = player.firstPlayed().get();
+        Instant lastPlayed = player.lastPlayed().get();
+        if (firstPlayed.equals(lastPlayed)) {
+            // Player has not been to this server before
```

Comment:
Yeah It's more of a footnote to the code-block. It's not really relevant to TextTemplates as a whole.